src/renderers/webgl/annotations/AnnotationsRenderer.ts

```diff
@@ -0,0 +1,65 @@
+import { Renderer } from '..'
+import { Annotation } from '../../../types'
+import CircleAnnotationRenderer from './CircleAnnotationRenderer'
+import LineAnnotationRenderer from './LineAnnotationRenderer'
+import RectangleAnnotationRenderer from './RectangleAnnotationRenderer'
+
+type AnnotationRendererLookup = Record<string, CircleAnnotationRenderer | RectangleAnnotationRenderer | LineAnnotationRenderer>
+
+export default class AnnotationsRenderer {
+  private annotations: Annotation[] = []
+  private lookup: AnnotationRendererLookup = {}
+
+  constructor(private renderer: Renderer) {
+    this.renderer = renderer
+  }
+
+  update(annotations: Annotation[] = []) {
+    if (this.annotations === annotations || (this.annotations.length === 0 && annotations.length === 0)) {
+      return this
+    }
+
+    const lookup: AnnotationRendererLookup = {}
+
+    for (const annotation of annotations) {
+      const renderer = this.lookup[annotation.id]
+
+      if (annotation.type === 'circle' && renderer instanceof CircleAnnotationRenderer) {
+        renderer.update(annotation)
+      } else if (annotation.type === 'circle') {
+        renderer?.delete()
+        lookup[annotation.id] = new CircleAnnotationRenderer(this.renderer, annotation)
+      } else if (annotation.type === 'rectangle' && renderer instanceof RectangleAnnotationRenderer) {
+        renderer.update(annotation)
+      } else if (annotation.type === 'rectangle') {
+        renderer?.delete()
+        lookup[annotation.id] = new RectangleAnnotationRenderer(this.renderer, annotation)
+      } else if (annotation.type === 'line' && renderer instanceof LineAnnotationRenderer) {
+        renderer.update(annotation)
+      } else if (annotation.type === 'line') {
+        renderer?.delete()
+        lookup[annotation.id] = new LineAnnotationRenderer(this.renderer, annotation)
+      }
+    }
+
+    for (const annotation of this.annotations) {
+      if (lookup[annotation.id] === undefined && this.lookup[annotation.id]) {
```

Comment:
shouldn't the second `this.lookup[annotation.id]` condition not be necessary? e.g.
```ts
for (const annotation of this.annotations) {
  if (lookup[annotation.id] === undefined) {
    annotation.delete()
  }
}
```
?